### PR TITLE
refactor event handlers

### PR DESCRIPTION
**Pull request by KipSigei**
Signed-off-by: Kipchirchir Sigei <arapgodsmack@gmail.com>

### CODE REVIEW

File: src/lib/components/Map/Map.js [hunk 1]
```diff
@@ -114,18 +114,11 @@ class Map extends Component {
...
-            // If it is a grouped layer loop through the layer.layers array
-            for (let l = 0; l < handler.layer.layers.length; l += 1) {
-              sublayer = handler.layer.layers[l];
-              this.map.on(handler.type, sublayer, handler.method);
-            }
-          } else {
-            this.map.on(handler.type, handler.layer.id, handler.method);
+        if (Array.isArray(handler.layer)) {
+          for (let l = 0; l < handler.layer.length; l += 1) {
+            this.map.on(handler.type, handler.layer[l], handler.method)
```
Comment (cKellyDesign) [inline src/lib/components/Map/Map.js]: @KipSigei is this working with grouped layers as well?

Comment (KipSigei) [reply src/lib/components/Map/Map.js]: At the moment it works by adding the sublayer ID's in the handler's layers array

Comment (cKellyDesign) [reply src/lib/components/Map/Map.js]: @kahama94 - can you make sure this change won't break anything in onaio/reveal-frontend#109

Comment (kahama94) [reply src/lib/components/Map/Map.js]: The change works fine

**Pull request merged**